### PR TITLE
F combined robot hw

### PR DESCRIPTION
**Pull request by dg-shadow**
Do not merge until changes have been made to HandE and HandH code to work with new system!

### CODE REVIEW

File: sr_ur_robot_hw/config/left_ur_arm_controller.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
-la_sr_ur_controller:
-  type: sr_ur_controller/UrArmController
+la_sr_ur_robot_hw:
```
Comment (toliver) [inline sr_ur_robot_hw/config/left_ur_arm_controller.yaml]: This file should be renamed as `la_sr_ur_robot_hw` is not a controller.

File: sr_ur_robot_hw/config/right_ur_arm_controller.yaml [hunk 2]
```diff
@@ -1,5 +1,5 @@
-ra_sr_ur_controller:
```
Comment (toliver) [inline sr_ur_robot_hw/config/right_ur_arm_controller.yaml]: File should be renamed.

File: sr_ur_bringup/launch/sr_ur_left.launch [hunk 3]
```diff
@@ -8,10 +8,10 @@
   <param name="robot_description" command="$(find xacro)/xacro.py '$(find sr_ur_description)/urdf/biarm.urdf.xacro'"/>
   
   <!-- left arm controller -->
-  <rosparam command="load" file="$(find sr_ur_controller)/config/left_ur_arm_controller.yaml"/>
+  <rosparam command="load" file="$(find sr_ur_robot_hw)/config/left_ur_arm_controller.yaml"/>
   <rosparam command="load" file="$(find sr_robot_launch)/config/la_trajectory_controller.yaml"/>
 
-  <node name="arm_controller_spawner" pkg="controller_manager" type="spawner" output="screen" args="la_sr_ur_controller"/>
+  <node name="arm_controller_spawner" pkg="controller_manager" type="spawner" output="screen" args="la_sr_ur_robot_hw"/>
```
Comment (toliver) [inline sr_ur_bringup/launch/sr_ur_left.launch]: Delete this. `la_sr_ur_robot_hw` is not a controller now.

File: sr_ur_bringup/launch/sr_ur_system.launch [hunk 4]
```diff
@@ -8,12 +8,12 @@
   <param name="robot_description" command="$(find xacro)/xacro.py '$(find sr_ur_description)/urdf/biarm.urdf.xacro'"/>
 
   <!-- low level controllers for right and left arm and a higher level trajectory controller -->
-  <rosparam command="load" file="$(find sr_ur_controller)/config/left_ur_arm_controller.yaml"/>
-  <rosparam command="load" file="$(find sr_ur_controller)/config/right_ur_arm_controller.yaml"/>
+  <rosparam command="load" file="$(find sr_ur_robot_hw)/config/left_ur_arm_controller.yaml"/>
+  <rosparam command="load" file="$(find sr_ur_robot_hw)/config/right_ur_arm_controller.yaml"/>
```
Comment (toliver) [inline sr_ur_bringup/launch/sr_ur_system.launch]: These 2 files should be renamed.

File: sr_ur_bringup/launch/sr_ur_right.launch [hunk 5]
```diff
@@ -8,10 +8,10 @@
   <param name="robot_description" command="$(find xacro)/xacro.py '$(find sr_ur_description)/urdf/biarm.urdf.xacro'"/>
   
   <!-- right arm controller -->
-  <rosparam command="load" file="$(find sr_ur_controller)/config/right_ur_arm_controller.yaml"/>
+  <rosparam command="load" file="$(find sr_ur_robot_hw)/config/right_ur_arm_controller.yaml"/>
```
Comment (toliver) [inline sr_ur_bringup/launch/sr_ur_right.launch]: These files should be renamed.

File: sr_ur_bringup/launch/sr_ur_left.launch [hunk 6]
```diff
@@ -8,10 +8,10 @@
   <param name="robot_description" command="$(find xacro)/xacro.py '$(find sr_ur_description)/urdf/biarm.urdf.xacro'"/>
   
   <!-- left arm controller -->
-  <rosparam command="load" file="$(find sr_ur_controller)/config/left_ur_arm_controller.yaml"/>
+  <rosparam command="load" file="$(find sr_ur_robot_hw)/config/left_ur_arm_controller.yaml"/>
```
Comment (toliver) [inline sr_ur_bringup/launch/sr_ur_left.launch]: These file should be renamed.

File: sr_ur_bringup/launch/sr_ur_right.launch [hunk 7]
```diff
@@ -8,10 +8,10 @@
...
   <rosparam command="load" file="$(find sr_robot_launch)/config/ra_trajectory_controller.yaml"/> 
 
-  <node name="arm_controller_spawner" pkg="controller_manager" type="spawner" output="screen" args="ra_sr_ur_controller"/>
+  <node name="arm_controller_spawner" pkg="controller_manager" type="spawner" output="screen" args="ra_sr_ur_robot_hw"/>
```
Comment (toliver) [inline sr_ur_bringup/launch/sr_ur_right.launch]: Delete this. `ra_sr_ur_robot_hw` is not a controller now.

File: sr_ur_launch/README.md [hunk 8]
```diff
@@ -0,0 +1,138 @@
+# Starting the robots (simulated / real)
```
Comment (toliver) [inline sr_ur_launch/README.md:1]: This file is basically describing how to use hand E and hand E + arm.
We should modify it to also contain instructions to launch UR arm alone.

File: sr_ur_launch/config/configuration.yaml [hunk 9]
```diff
@@ -0,0 +1,39 @@
+robot_hardware:
+  - ra_sr_ur_robot_hw
+
+ra_sr_ur_robot_hw:
```
Comment (toliver) [inline sr_ur_launch/config/configuration.yaml]: This is duplicating the information in right_ur_arm_controller.yaml. Maybe that file should be removed?

File: sr_ur_launch/launch/right_arm.launch [hunk 10]
```diff
@@ -0,0 +1,23 @@
+<launch>
```
Comment (toliver) [inline sr_ur_launch/launch/right_arm.launch]: This file looks like a replacement for:

sr_ur_bringup/launch/sr_ur_right.launch

Should that one be deleted?

And the left arm one as well?

sr_ur_bringup/launch/sr_ur_left.launch

File: sr_ur_launch/scripts/teach_mode_node [hunk 11]
```diff
@@ -0,0 +1,245 @@
...
+
+class ControllerHelper(object):
+
+    def __init__(self, robot_ids, robot_joint_prefixes):
+        self.time_to_reload_params = 6.0
+        self.robot_ids = robot_ids
+        self.robot_joint_prefixes = robot_joint_prefixes
+
+        # This section will only be used for hands
+        self.joints = ["ffj0", "ffj3", "ffj4",
```
Comment (toliver) [inline sr_ur_launch/scripts/teach_mode_node:20]: We will have to adapt this to be compatible with hand H. We need to add this to our Jira backlog.

File: sr_ur_robot_hw/CMakeLists.txt [hunk 12]
```diff
@@ -1,42 +1,46 @@
...
 	    controller_interface
+	    hardware_interface
 	    ros_ethercat_model
 	    sr_ur_msgs
 	    libuv-dev
 	INCLUDE_DIRS
 	    include
 	LIBRARIES
-		sr_ur_controller
+            sr_ur_robot_hw
```
Comment (toliver) [inline sr_ur_robot_hw/CMakeLists.txt]: Fix indentation.

**Review by toliver — CHANGES_REQUESTED** [review]

File: sr_ur_launch/launch/right_arm.launch (hunk 10 above)
Comment (dg-shadow) [reply sr_ur_launch/launch/right_arm.launch]: There is no left_arm one... We could either move both from sr_ur_bringup or just leave them there...

TBH, nothing anywhere I can find even uses sr_ur_launch for anything... we could just delete the whole package I think.

File: sr_ur_robot_hw/CMakeLists.txt (hunk 12 above)
Comment (dg-shadow) [inline sr_ur_robot_hw/CMakeLists.txt]: ?? 
Indentation looks correct...

**Review by toliver — APPROVED** [review]

**Pull request merged**